src/mlpack/tests/lmnn_test.cpp

```diff
@@ -188,18 +185,18 @@ BOOST_AUTO_TEST_CASE(LMNNInitialEvaluateWithGradientTest)
   double objective = lmnnfn.EvaluateWithGradient(coordinates, gradient);
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(objective, 9.456, 1e-5);
+  REQUIRE(objective == Approx( 9.456).epsilon(1e-7));
   // Check Gradient
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.288, 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(1, 0), 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(0, 1), 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 12.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.288).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(0, 1) == Approx(0.0).margin(1e-5));
+  REQUIRE(gradient(1, 1) == Approx( 12.0).epsilon(1e-7));
```

Comment:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.288).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).margin(1e-5));
  REQUIRE(gradient(0, 1) == Approx(0.0).margin(1e-5));
  REQUIRE(gradient(1, 1) == Approx(12.0).epsilon(1e-7));
```

Remove the extra space.